test/integration/targets/copy/tasks/main.yml

```diff
@@ -106,6 +106,42 @@
           - '"Ansible managed" in diff_output.diff[0].after'
           - '"file.txt" in diff_output.diff[0].after_header'
 
+    - name: create new directory
+      file:
+        path: "{{ role_path }}/files/two-file-diff-test.d"
+        state: directory
```

Comment:
Please add `delegate_to: localhost`, to ensure this directory is local (or you could add to test/integration/targets/copy/files/ non-dynamically, that's probably clearer), and ensure the dir contains a dir, file, and symlink to exercise each of the codepaths you fixed.

Reply:
all set. When doing this I ran into a funny edge case where the symlink diff clobbers the file and directory diffs, but only when there is exactly 1 file copied. Also noticed that when copying a blank file, the result is `before="" after=""`, which probably should be `before=None after=""`, or another diff should be returned saying `state=absent -> state=present`.